File: src/types/models/sensors.ts
export interface SensorsWindowsSensor {
  id: string;
  name: string;
  type: string;
  value?: number | string;
}

export interface SensorsWindowsHardware {
  id: string;
  name: string;
  type: string;
  subhardware: SensorsWindowsHardware[];
  sensors: SensorsWindowsSensor[];
}

export interface SensorsWindows {
  hardware?: SensorsWindowsHardware[];
  nvidia?: any;
}

export interface Sensors {
  fans?: any;
  temperatures?: any;
  windows_sensors?: SensorsWindows;
}

File: backend/data/module/sensors/sensors.go
package sensors

import "github.com/timmo001/system-bridge/types"

func GetSensorsData() (types.SensorsData, error) {
	var sensorsData types.SensorsData
	// Initialize arrays
	sensorsData.WindowsSensors = &types.SensorsWindows{
		Hardware: make([]types.SensorsWindowsHardware, 0),
		NVIDIA: &types.SensorsNVIDIA{
			Displays: make([]types.SensorsNVIDIADisplay, 0),
			GPUs:     make([]types.SensorsNVIDIAGPU, 0),
		},
	}

	// TODO: Implement
	return sensorsData, nil
}
